### ISSUE TITLE
Create categorical contingency table data structure

### ISSUE THREAD

**Issue by fernando-aristizabal** [NOAA-OWP/gval#17]
Implement a standard schema for contingency table structure.

This issue depends on #38.

## Current behavior
- Some post processing is implemented to index cross-tabulated counts by unique values in candidate map (rows) and benchmark map (columns).
- No ability to keep track of candidate or benchmark attributes or process parameters is in place.

## Expected behavior

- The above structure is great for readability but does not account for the various hierarchies of samples including sub-samples, band/features, maps, and catalogs.
- A contingency table structure is required that reports cross-tabulated counts for each sample as well as their associated attributes.
- Methods to aggregate/groupby crosstab counts by level or associated attributes is important.
- Ways of melting and pivoting this structure to something more human readable like a cross-tabulation table indexed by unique values in candidate map (rows) and benchmark map (columns) would be nice to have as well.
- Some method of tagging each column with metadata is necessary. Current metadata source (candidate, benchmark, or process) and hierarchy (catalog, map, band/feature, sub-sample).
    - Consider this approach for [multi-indexing](https://notes.mikejarrett.ca/storing-metadata-in-pandas-dataframes/).

- What data structures are necessary to account for this?
- What classes should be created for this?
